docs/sources/experiments/algorithm.md

```diff
@@ -310,7 +310,18 @@ The trained model's prediction probabilities (`predict_proba()`) are computed bo
+Triage allows for the computation of both testing set and training set evaluation metrics. Evaluation metrics, such as precision and recall at various thresholds, are written to either the `train_results.evaluations` table or the `test_results.evaluations`. Triage defines a number of [Evaluation Metrics](https://github.com/dssg/triage/blob/master/src/triage/component/catwalk/evaluation.py#L45-L58) metrics that can be addressed by name in the experiment definition, along with a list of thresholds and/or other parameters (such as the 'beta' value for fbeta) to iterate through.
+
+Thresholding is done either via absolute value (top k) or percentile by sorting the predictions and labels by the row's predicted probability score, with ties broken in some way (see next paragraph), and assigning the predicted value as True for those above the threshold. Note that the percentile thresholds are in terms of the population percentage, not a cutoff threshold for the predicted probability.
+
+A few different versions of tiebreaking are implemented to deal with the nuances of thresholding, and each result is written to the evaluations table for each metric score, along with some related statistics:
+
+* `worst_value` - Ordering by the label ascending. This has the effect of as many predicted negatives making it above thresholds as possible, thus producing the worst possible score.
+* `best_value` - Ordering by the label descending. This has the effect of as many predicted positives making it above thresholds as possible, thus producing the best possible score.
+* `stochastic_value` - If the `worst_value` and `best_value` are not the same (as defined by the floating point tolerance at catwalk.evaluation.RELATIVE_TOLERANCE), the sorting/thresholding/evaluation will be redone many times, and the mean of all these trials is written to this column. Otherwise, the `worst_value` is written here
+* `num_sort_trials` - If trials are needed to produce the `stochastic_value`, the number of trials taken is written here. Otherwise this will be 1
+* `standard_deviation` - If trials are needed to produce the `stochastic_value`, the standard deviation of these trials is written here. Otherwise this will be null
```

Comment:
I don't have a strong feeling on this, but perhaps the default for st dev should be 0 (i.e., no variance).

Reply:
From the error we get in both Python and Postgres, it seems to be 'undefined' with less than 2 trials and I call this different than 0. 

On the other hand, if we take my answer to the other comment (the best and/or worst is a trial) we could just plug the best and worst in as the two trials, which is a defined variance of 0.

Reply:
Yeah, I guess the difference is: do we take license to say we know the std dev of the metric is 0 when all values are (within our sig digits) equal, even if we don't do the trials, or do we report the estimated standard deviation only when we've estimated it with trials.

Reply:
In other words: Just like the mean reported in `stochastic_value` is an estimate of the true mean of all possible metric values, the standard deviation based on the trials is just an estimate of the true standard deviation of all possible values. When all of the possible values are the same, we know the population value is 0 without having to do the trials, so we could just enter that.